x-pack/plugins/security/server/authorization/api_authorization.ts

```diff
@@ -47,26 +57,61 @@ export function initAPIAuthorization(
 
       const authz = security.authz as AuthzEnabled;
 
-      const requestedPrivileges = authz.requiredPrivileges.flatMap((privilegeEntry) => {
-        if (typeof privilegeEntry === 'object') {
-          return [...(privilegeEntry.allRequired ?? []), ...(privilegeEntry.anyRequired ?? [])];
+      const { requestedPrivileges, requestedReservedPrivileges } = authz.requiredPrivileges.reduce(
+        (acc, privilegeEntry) => {
+          const privileges =
+            typeof privilegeEntry === 'object'
+              ? [...(privilegeEntry.allRequired ?? []), ...(privilegeEntry.anyRequired ?? [])]
+              : [privilegeEntry];
+
+          for (const privilege of privileges) {
+            if (isReservedPrivilegeSet(privilege)) {
+              acc.requestedReservedPrivileges.push(privilege);
+            } else {
+              acc.requestedPrivileges.push(privilege);
+            }
+          }
+
+          return acc;
+        },
+        {
+          requestedPrivileges: [] as string[],
+          requestedReservedPrivileges: [] as string[],
         }
+      );
 
-        return privilegeEntry;
-      });
-
-      const apiActions = requestedPrivileges.map((permission) => actions.api.get(permission));
       const checkPrivileges = checkPrivilegesDynamicallyWithRequest(request);
-      const checkPrivilegesResponse = await checkPrivileges({ kibana: apiActions });
+      const checkPrivilegesIfNotEmpty = async () => {
+        if (requestedPrivileges.length === 0) {
+          return;
+        }
+
+        const apiActions = requestedPrivileges.map((permission) => actions.api.get(permission));
+
+        return await checkPrivileges({ kibana: apiActions });
+      };
 
       const privilegeToApiOperation = (privilege: string) =>
         privilege.replace(API_OPERATION_PREFIX, '');
+
+      const checkPrivilegesResponse = await checkPrivilegesIfNotEmpty();
       const kibanaPrivileges: Record<string, boolean> = {};
 
-      for (const kbPrivilege of checkPrivilegesResponse.privileges.kibana) {
+      for (const kbPrivilege of checkPrivilegesResponse?.privileges?.kibana ?? []) {
         kibanaPrivileges[privilegeToApiOperation(kbPrivilege.privilege)] = kbPrivilege.authorized;
       }
 
+      for (const reservedPrivilege of requestedReservedPrivileges) {
```

Comment:
>[!NOTE]
> For now, we don’t need to do anything, but if we decide to optimize performance in the future, it would make sense to make only one call first (e.g., for superuser privileges) and see if we can return a 403 early. Then, we can make the second call for the remaining privileges group, and so on.

Reply:
I guess we allow to use superuser privileges as a single entry option, this won't be needed